resharper/resharper-unity/src/Rider/MonoInstallTrigger.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Linq;
+using JetBrains.Application.Threading;
+using JetBrains.Lifetimes;
+using JetBrains.ProjectModel;
+using JetBrains.ProjectModel.Tasks;
+using JetBrains.ReSharper.Host.Features.Runtime;
+using JetBrains.ReSharper.Plugins.Unity.ProjectModel;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Rider
+{
+    [SolutionComponent]
+    public class MonoInstallTrigger
+    {
+        public MonoInstallTrigger(Lifetime lifetime, ILogger logger, ISolutionLoadTasksScheduler scheduler,
+            ISolution solution, UnitySolutionTracker unitySolutionTracker, UnityHost host)
+        {
+            if (PlatformUtil.RuntimePlatform != PlatformUtil.Platform.MacOsX)
+                return;
+
+            scheduler.EnqueueTask(new SolutionLoadTask("Check mono runtime", SolutionLoadTaskKinds.AfterDone, () =>
+            {
+                if (!unitySolutionTracker.IsUnityGeneratedProject.Value)
+                    return;
+
+                if (!HasModernUnityProjects(solution))
+                    return;
+
+                solution.Locks.Tasks.Queue(lifetime, () =>
+                {
+                    var wellKnownMonoRuntimes = MonoRuntimeDetector.DetectWellKnownMonoRuntimes();
+                    var installedValidMono = wellKnownMonoRuntimes
+                        .Any(runtime =>
+                        {
+                            var version = new Version(0, 0); // if we fail to parse version - consider it is old
+                            try
+                            {
+                                version = new Version(ProcessOutputUtil.ExtractMonoVersion(runtime.ExePath));
+                            }
+                            catch (Exception e)
+                            {
+                                logger.Warn(e);
+                            }
+
+                            return version.Major >= 5 && version.Minor >= 16;
```

Comment:
Can you add a comment why it has to be >= 5.16?

Reply:
done